server/src/main/java/de/uftos/repositories/solver/timefold/domain/LessonTimefoldInstance.java

```diff
@@ -0,0 +1,92 @@
+package de.uftos.repositories.solver.timefold.domain;
+
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
+import ai.timefold.solver.core.api.domain.lookup.PlanningId;
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+import com.fasterxml.jackson.annotation.JsonIdentityReference;
+import de.uftos.dto.ResourceType;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * This class models a lesson that can be used by the Timefold solver.
+ */
+@Getter
+@Setter
+@PlanningEntity
+public class LessonTimefoldInstance implements ResourceTimefoldInstance {
+
+  @PlanningId
+  private String id;
+  private int index;
+  @JsonIdentityReference
+  @PlanningVariable
+  private TimeslotTimefoldInstance timeslot;
+  @JsonIdentityReference
+  @PlanningVariable
+  private TeacherTimefoldInstance teacher;
+  @JsonIdentityReference
+  private StudentGroupTimefoldInstance studentGroup;
+  @JsonIdentityReference
+  private SubjectTimefoldInstance subject;
+  @JsonIdentityReference
+  @PlanningVariable
+  private RoomTimefoldInstance room;
+
+  //required by the timefold solver
+  public LessonTimefoldInstance() {
+
+  }
```

Comment:
Replace this with lombok's `@NoArgsConstructor`

Reply:
I'll change it.